src/components/tree-views-container.js

```diff
@@ -522,43 +541,32 @@ const TreeViewsContainer = () => {
                     ) {
                         dispatch(setSelectedDirectory(null));
                     }
-                    return;
-                }
 
-                if (directoryUuid) {
-                    if (directoryUuid === selectedDirectoryRef.current) {
-                        updateDirectoryTreeAndContent(directoryUuid);
-                    } else {
-                        updateDirectoryTree(directoryUuid);
+                    if (directoryUuid) {
```

Comment:
Here if a directory which is not a root directory received an update notification you will not updateTree nor content.
You should do this I think (made in another PR :
```javascript
                if (isRootDirectory) {
                    updateRootDirectories();
                    if (
                        selectedDirectoryRef.current != null && // nothing to do if nothing already selected
                        notificationTypeHeader ===
                            notificationType.DELETE_DIRECTORY &&
                        selectedDirectoryRef.current.elementUuid ===
                            directoryUuid
                    ) {
                        dispatch(setSelectedDirectory(null));
                    }
                    return;
                }

                if (directoryUuid) {
                    // Remark : It could be a Uuid of a rootDirectory if we need to update it because its content update
                    // if dir is actually selected then call updateDirectoryTreeAndContent of this dir
                    // else expanded or not then updateDirectoryTree
                    if (selectedDirectoryRef.current != null) {
                        if (
                            directoryUuid ===
                            selectedDirectoryRef.current.elementUuid
                        ) {
                            updateDirectoryTreeAndContent(directoryUuid);
                            return; // break here
                        }
                    }
                    updateDirectoryTree(directoryUuid);
                }
```
I added a check on `if (selectedDirectoryRef.current != null) {` too, because renaming a rootDirectory with right click when nothing is selected caused a crash otherwise ;)

Reply:
Corrected